src/vs/editor/common/services/languageService.ts

```diff
@@ -150,51 +151,15 @@ export class LanguageService extends Disposable implements ILanguageService {
 }
 
 class LanguageSelection implements ILanguageSelection {
+	private readonly _value: IObservable<string>;
+	public readonly onDidChange: Event<string>;
 
-	public languageId: string;
-
-	private _listener: IDisposable | null = null;
-	private _emitter: Emitter<string> | null = null;
-
-	constructor(
-		private readonly _onDidChangeLanguages: Event<void>,
-		private readonly _selector: () => string
-	) {
-		this.languageId = this._selector();
-	}
-
-	private _dispose(): void {
-		if (this._listener) {
-			this._listener.dispose();
-			this._listener = null;
-		}
-		if (this._emitter) {
-			this._emitter.dispose();
-			this._emitter = null;
-		}
+	constructor(onDidChangeLanguages: Event<void>, selector: () => string) {
+		this._value = observableFromEvent(this, onDidChangeLanguages, () => selector());
+		this.onDidChange = Event.fromObservable(this._value);
```

Comment:
Do you not need to register this?

Reply:
Why? This is the event, not the event subscription. As for the Emitter, disposing it is optional and doesn't help here (which is easiest to see in the old implementation, where the emitter would only be disposed if the last listener unsubscribes, in which case disposing the emitter doesn't do anything).
Emitters are not considered by the disposable tracking tooling.